v2/jwt/error.go

```diff
@@ -0,0 +1,26 @@
+package jwt
+
+var ErrNotValidNow = errExpiredType{}
+
+func errNotValidNow(underlyingErr error) error {
+
+	return errExpiredType{
+		underLyingErr: underlyingErr,
+	}
+}
+
+type errExpiredType struct {
+	underLyingErr error
+}
+
+func (e errExpiredType) Error() string {
+	return "token is not valid right now: " + e.underLyingErr.Error()
+}
+
+func (e errExpiredType) Unwrap() error {
+	return e.underLyingErr
+}
+
+func (e errExpiredType) Is(_ error) bool {
+	return true
+}
```

Comment:
f we don't need this implemented, can't we just leave it out? Guessing errors.Is does some magic even if this returns true?

Reply:
Unless we implement it it will look for instance equality.
This way it is "is" if it is this type.